### ISSUE TITLE
Webpack Integration

### ISSUE THREAD

**Issue by aj-dev-smith** [amzn/style-dictionary#515]
Hi, first off, awesome work on this project. It has been extremely useful for my team and I!

I'm looking for more info on how the node api can be integrated with Webpack.  In the API for the [`exportPlatform` function](https://amzn.github.io/style-dictionary/#/api?id=exportplatform), it mentions that function can be useful in integrating with tools like webpack.  Is there anywhere I can find an example of this?

Currently, I'm working in a monorepo, using Style Dictionary as the source of truth for all tokens to be used across the system.  The current setup, just follows the theme example and uses `buildAllPlatforms()` plus a little extra to generate our vars.  Then we have to check the result into source control, which has led to much confusion and issues.

Ideally we'd want to update our webpack config to include the style dictionary build, to remove the need for checking in the generated files.  They'd be generated on the fly similar to the rest of the app.  I've looked into Webpack Loaders.  It doesn't seem like one would work because they only parse a single input file at a time, unlike Style Dictionary, which uses the whole `properties` directory as an input.  Possibly a custom Webpack Plugin could work, but that's a bit beyond my current knowledge of webpack.

Any help would be appreciated.  Thanks!

**Comment by dbanksdesign** [comment]
Sorry for the lateness of this response, just getting back from the holidays. I think you are right, a webpack loader doesn't seem like the proper place because as you mention style dictionary isn't really a 1:1 with input and output files. 

We do not currently have an example in this repository showing a webpack/monorepo integration. However, I gave a workshop last year that has similar set up with a token package and a webpack app consuming it (along with android and ios apps):

https://github.com/dbanksdesign/clarity-2020-design-token-workshop/tree/final

The top level directory is the token package, it has it's own package.json file that has the style dictionary build scripts. In the 'web' directory is a webpack application that depends on token package, named 'clarity-design-tokens'. I did not set it up as a monorepo just due to time constraints, but using `npm link` and `npm link clarity-design-tokens` to symlink the packages works  similarly to a monorepo setup. The token package builds files that the webpack app consumes. 

The one downside to this approach that I haven't found a way around is if you are actively editing tokens _and_ component/application code, you will need 2 terminal windows open, both with watcher processes. But one nice thing is that with both watcher processes, if you edit a token it will kick off the style dictionary build, which will generate a file that the webpack app is importing which will cause the webpack watcher to also rebuild. 

Here are some other mono repo set ups that might help:
* https://github.com/sproutsocial/seeds-packets/
* https://github.com/Skyscanner/backpack
